src/honey/sql.cljc

```diff
@@ -214,16 +215,16 @@
      (fn [fk _] (param-value (fk)))}))
 
 (defn- format-var [x & [opts]]
-  (let [c (name-_ x)]
+  (let [c (name x)]
     (cond (= \% (first c))
-          (let [[f & args] (str/split (subs c 1) #"\.")]
-            ;; TODO: this does not quote arguments -- does that matter?
-            [(str (upper-case f) "(" (str/join "," args) ")")])
+          (let [[f & args] (str/split (subs c 1) #"\.")
+                quoted-args (map #(format-entity (keyword %) opts) args)]
+            [(str (upper-case (-_ f)) "(" (str/join ", " quoted-args) ")")])
           (= \? (first c))
-          (let [k (keyword (subs c 1))]
-           (if *inline*
-             [(sqlize-value (param-value k))]
-             ["?" (->param k)]))
+          (let [k (keyword (subs (-_ c) 1))]
```

Comment:
`-_` is not needed here. You've actually uncovered -- and almost fixed -- a bug that has been lurking for a while.

`:?foo-bar` should map to `{:params {:foo-bar 123}}` in the named parameters passed to `format` but the code currently calls `name-_` which would turn this into `:foo_bar` (which is actually wrong). Thank you for highlighting that bug!

It does, however, suggest to me that there's a deeper bug lurking here as well -- not covered in the new tests -- that I mentioned this in a previous review comment: the tests need to verify that qualified column names work properly (which they won't, so that's going to lead to more work needed here).

I think at this point I'm going to take the PR as-is and make the changes mentioned in the review, and beef up the tests and figure out how to fix the qualified column bug that I can now see exists in the code.